elliottlib/cli/verify_attached_operators_cli.py

```diff
@@ -151,27 +191,48 @@ def _extract_available_image_shasums(image_builds):
+        for build in bundles
+        for ref in build['csv']['spec']['relatedImages']
+    ]
+    green_print(f"Found {len(bundles)} bundles with {len(references)} references")
     missing = set()
-    for image_pullspec, metadata in references.items():
-        digest = image_pullspec.split("@")[1]  # just the shasum
-        if digest in available:
-            continue
-        ref = image_pullspec.rsplit('/', 1)[1]  # cut off the registry/namespace, just need the name:shasum
-        res = ref
+    for image_pullspec, build in references:
+        # validate an image reference from a bundle is shipp(ed/ing) to the right repo
+        repo, digest = image_pullspec.rsplit("@", 1)  # split off the @sha256:...
+        _, repo = repo.split("/", 1)  # pick off the registry
+        ref = image_pullspec.rsplit('/', 1)[1]  # just need the name@digest
+
         try:
-            nvr = _nvr_for_operand_pullspec(runtime, ref)
-            build = brew.get_brew_build(nvr=nvr)
-            if [ad for ad in build.all_errata if ad["status"] != "DROPPED_NO_SHIP"]:
-                continue
+            ref = _nvr_for_operand_pullspec(runtime, ref)  # convert ref to nvr
+            context = f"Bundle {build['nvr']} reference {ref}:\n   "
+            attached_advisories = _get_attached_advisory_ids(ref)
+            cdn_repos = _get_cdn_repos(attached_advisories, ref)
+            if digest not in available_shasums and not attached_advisories:
+                red_print(f"{context} not shipped or attached to any advisory.")
+            elif not cdn_repos:
+                red_print(f"{context} does not have any CDN repos on advisory it is attached to")
+            elif repo not in cdn_repos:
+                red_print(f"{context} needs CDN repo '{repo}' but advisory only has {cdn_repos}")
+            elif digest in available_shasums:
+                green_print(f"{context} shipped/shipping as {image_pullspec}")
+                continue  # do not count it as missing
+            elif omit_attached:
+                # not already shipped (or cmdline omitted shipped), nor in a listed advisory;
+                # if we passed above gates, it is attached to some other advisory;
+                # but cmdline option says to count that as missing.
+                red_print(f"{context} only found in omitted advisory {attached_advisories}")
             else:
-                res = nvr
+                green_print(f"{context} attached to separate advisory {attached_advisories}")
+                continue  # do not count it as missing
         except BrewBuildException:
-            # Fall through to missing.add
-            pass
-        missing.add(res)
-        red_print(f"{metadata} has a reference to {res} not present in the advisories nor shipped images.")
+            # advisory lookup for brew build failed, fall through to count as missing
+            red_print(f"{context} failed to look up in errata-tool")
```

Comment:
`BrewBuildException` will also be raised in case of connection/auth error. It would be nice to print the error out.

Reply:
Is it? I thought that would be a different exception. Good idea to at least print it out.